firmware/firmware.h

```diff
@@ -90,6 +91,17 @@ void bridge_close_3();
 
 // port.c
 
+#define UART_MS_TIMEOUT 10 // send uart data after ms timeout even if buffer is not full
+#define UART_RX_SIZE 30
+#define RX_BUF_INCR(pos)  (pos=((++pos) == (UART_RX_SIZE - 1) ? 0 : pos))
```

Comment:
This would be better written as `pos = (pos + 1) % UART_RX_SIZE`, which if your UART_RX_SIZE is a power of two, will optimize into a bitwise AND.